duet/api.py

```diff
@@ -418,9 +418,10 @@ async def __aexit__(self, exc_type, exc, tb) -> None:
 
     def _release(self):
         self._count -= 1
-        if self._waiters:
+        while self._waiters:
```

Comment:
nit -- add a comment here, something like "Release the first non-cancelled and valid waiter"